Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -815,6 +815,44 @@ void LocalFrameViewLayoutContext::checkLayoutState()
 }
 #endif
 
+AnchorScrollAdjuster* LocalFrameViewLayoutContext::anchorScrollAdjusterFor(const RenderBox& anchored)
+{
+    auto index = m_anchorScrollAdjusters.findIf([&](auto& item) {
+        return item.anchored() == &anchored;
+    });
+    if (WTF::notFound == index)
+        return { };
+    return &m_anchorScrollAdjusters[index];
+}
+
+void LocalFrameViewLayoutContext::registerAnchorScrollAdjuster(AnchorScrollAdjuster&& scrollAdjuster)
+{
+
+    auto index = m_anchorScrollAdjusters.findIf([&](auto& item) {
+        return item.anchored() == scrollAdjuster.anchored();
+    });
+    if (WTF::notFound == index)
+        m_anchorScrollAdjusters.append(WTFMove(scrollAdjuster));
+    else
+        m_anchorScrollAdjusters[index] = WTFMove(scrollAdjuster);
+}
+
+void LocalFrameViewLayoutContext::unregisterAnchorScrollAdjusterFor(const RenderBox& anchored)
+{
+    m_anchorScrollAdjusters.removeFirstMatching([&](auto& item) {
+        return item.anchored() == &anchored;
+    });
+    ASSERT(!m_anchorScrollAdjusters.containsIf([&](auto& item) {
```

Comment:
You should use a data structure that doesn't allow duplicates (HashSet, HashMap) instead of asserting.

Reply:
I wanted to keep these items inline in memory rather than externalizing them on the heap, and it doesn't seem those data types allow for reference access to inline data.